### ISSUE TITLE
Be more verbose when failing to fetch metadata 

### ISSUE THREAD

**Issue by fee-mendes** [scylladb/scylla-rust-driver#1021]
As seen in https://github.com/scylladb/scylladb/issues/19107, sometimes the driver may fail to fetch metadata from its peers and print messages in the logs such as:

```rust
2024-06-24T19:05:16.729306Z  WARN scylla::transport::topology: Failed to establish control connection and fetch metadata on all known peers. Falling back to initial contact points.
2024-06-24T19:05:16.729347Z  WARN scylla::transport::topology: Failed to fetch metadata using current control connection control_connection_address="172.31.29.124:9042" error=Protocol Error: system.peers or system.local has invalid column type
2024-06-24T19:05:16.733118Z ERROR scylla::transport::topology: Could not fetch metadata error=Protocol Error: system.peers or system.local has invalid column type
```

It is wise to know exactly which invalid data was found, such as from which `host_id`, which column was found to be invalid and which value was returned. 

Similar work has been done for the Python driver in https://github.com/scylladb/python-driver/issues/303

**Comment by mykaul** [comment]
The root cause might be due to https://github.com/scylladb/scylla-rust-driver/issues/1023 btw.

**Comment by dkropachev** [comment]
@fee-mendes, original problem is solved in https://github.com/scylladb/scylla-rust-driver/issues/1023, and verbose logging is going to be implemented at https://github.com/scylladb/scylla-rust-driver/issues/462

**Comment by wprzytula** [comment]
This has been included in release 0.13.2.
